Add cause of JsonMappingException to failureMessage on VariableAssertj.hasVariableSatisfies

### <!-- Component -->

<!-- Camunda Process Test- -->

### User Story

In case deserializing an object variable value to a class fails, the underlying json parsing error is currently not transparent to the user.

e.g.
```
assertThat(pi).hasVariableSatisfies("object", MyClass.class, object -> {
Assertions.assertThat(object.getId()).isNotNull();
});
```
can fail with
```
java.lang.AssertionError: Process instance [key: 2251799813685293] should have a variable 'object' of type 'MyClass', but was: '"{\"id\":1"}"'
```
while the underlying error would be more helpful to pinpoint the cause (here, the value being a string)
```
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `MyClass` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('{"id":1}')
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 1]
```

### Design

Ideally we include the cause of the JsonMappingException in the failureMessage.

### Technical requirements

_No response_

### Dev -> QA handover

- Resources:
- Versions to validate:
- Release version (in which version this feature will be released):


### QA test case

_No response_

### Links

https://jira.camunda.com/browse/SUPPORT-28814

--- a/testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/assertions/VariableAssertj.java
+++ b/testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/assertions/VariableAssertj.java
@@ -214,8 +214,14 @@ public class VariableAssertj extends AbstractAssert<VariableAssertj, String> {
           } catch (final JsonMappingException e) {
             final String failureMessage =
                 String.format(
-                    "%s should have a variable '%s' of type '%s', but was: '%s'",
-                    actual, variableName, variableValueType.getName(), actualVariable);
+                    "%s should have a variable '%s' of type '%s', but the JSON mapping failed:\n"
+                        + "Error: %s\n"
+                        + "Reason: %s",
+                    actual,
+                    variableName,
+                    variableValueType.getName(),
+                    e.getMessage(),
+                    e.getCause());
 
             fail(failureMessage);
           }

--- a/testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/assertions/VariableAssertj.java
+++ b/testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/assertions/VariableAssertj.java
@@ -212,16 +212,17 @@ public class VariableAssertj extends AbstractAssert<VariableAssertj, String> {
                 actual, variableName, e.getMessage());
 
           } catch (final JsonMappingException e) {
+            final Throwable reason =
+                Optional.ofNullable(e.getCause())
+                    .map(cause -> Optional.ofNullable(cause.getCause()).orElse(cause))
+                    .orElse(e);
+
             final String failureMessage =
                 String.format(
                     "%s should have a variable '%s' of type '%s', but the JSON mapping failed:\n"
                         + "Error: %s\n"
                         + "Reason: %s",
-                    actual,
-                    variableName,
-                    variableValueType.getName(),
-                    e.getMessage(),
-                    e.getCause());
+                    actual, variableName, variableValueType.getName(), e.getMessage(), reason);
 
             fail(failureMessage);
           }

--- a/testing/camunda-process-test-java/src/test/java/io/camunda/process/test/api/VariableAssertTest.java
+++ b/testing/camunda-process-test-java/src/test/java/io/camunda/process/test/api/VariableAssertTest.java
@@ -777,10 +777,10 @@ public class VariableAssertTest {
                           COMPLEX_VARIABLE_KEY,
                           List.class,
                           result -> Assertions.assertThat(result).hasSize(5)))
-          .hasMessage(
-              "Process instance [key: 1] should have a variable 'complex' of type 'java.util.List', but was: '"
-                  + jsonMapper.readJson(COMPLEX_VARIABLE_VALUE)
-                  + "'");
+          .hasMessageContainingAll(
+              "Process instance [key: 1] should have a variable 'complex' of type 'java.util.List', but the JSON mapping failed:\n",
+              "Error: Failed to read JSON: '" + COMPLEX_VARIABLE_VALUE + "'\n",
+              "Reason: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.util.ArrayList<java.lang.Object>` from Object value (token `JsonToken.START_OBJECT`)");
     }
 
     @Test

